### PR TITLE
CFN Template added

### PR DESCRIPTION
**Pull request by ngoyal16**
**Description of changes:**

CFN Template added so that end users can easily create the queue and event subscriptions

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/cfn-template.yaml [hunk 1]
```diff
@@ -0,0 +1,68 @@
+AWSTemplateFormatVersion: 2010-09-09
+Resources:
```
Comment (AustinSiu) [inline docs/cfn-template.yaml:2]: Can you add a rule for AWS Health Events? It should look something like this:

```
  ScheduledChangeRule:
    Type: 'AWS::Events::Rule'
    Properties:
      EventPattern:
        source:
          - aws.health
        detail-type:
          - AWS Health Event
      Targets:
        - Id: 1
          Arn: !GetAtt Queue.Arn
```

File: docs/cfn-template.yaml [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    Properties:
+      EventPattern:
+        source:
+          - aws.ec2
+        detail-type:
+          - EC2 Instance State-change Notification
+      Targets:
+        - Id: 1
+          Arn: !GetAtt Queue.Arn
+Outputs:
```
Comment (AustinSiu) [inline docs/cfn-template.yaml:65]: Can you please update the sections of the "[Infrastructure Setup](https://github.com/aws/aws-node-termination-handler#infrastructure-setup)" instructions that this template covers? It looks like you're capturing Step 1 and Step 3.

**Review by AustinSiu — CHANGES_REQUESTED** [review]
Thank you for your work on this @ngoyal16!
Just a few comments, please take a look.

File: docs/cfn-template.yaml (hunk 1 above)
Comment (ngoyal16) [reply docs/cfn-template.yaml:2]: Rule has been added

File: docs/cfn-template.yaml (hunk 2 above)
Comment (ngoyal16) [reply docs/cfn-template.yaml:76]: @AustinSiu should remove cli commands for SQS and EventBridge Creation?

Comment (AustinSiu) [reply docs/cfn-template.yaml:76]: We can leave the cli command references in the docs, for convenience. I was thinking more of adding a sentence pointing the reader to the template in the relevant. Something like the below to replace [this](https://github.com/aws/aws-node-termination-handler#1-create-an-sqs-queue):
```
Here is the AWS CLI command to create an SQS queue to hold termination events from ASG and EC2, although this should really be configured via your favorite infrastructure-as-code tool like CloudFormation or Terraform (template describing these resources can be found [here](url)):
```

Comment (ngoyal16) [reply docs/cfn-template.yaml:76]: @AustinSiu changes are made as per request.

Anything else needs to change let me know.

Comment (AustinSiu) [reply docs/cfn-template.yaml:76]: Thanks! Can you add it to "4. Create Amazon EventBridge Rules" as well?

**Comment by AustinSiu** [comment]
The failing tests are not related to your changes, so don't be concerned about that. We'll get a PR out this afternoon to fix it. Similar change can be seen here: https://github.com/aws/amazon-ec2-metadata-mock/pull/168

File: docs/cfn-template.yaml (hunk 2 above)
Comment (ngoyal16) [reply docs/cfn-template.yaml:76]: @AustinSiu done

**Comment by AustinSiu** [comment]
@ngoyal16 we [merged a fix](https://github.com/aws/aws-node-termination-handler/commit/e3258a8f60f5f58e6d1ddbbcb0a654d83842ecb4) for the failing tests, thanks for your patience. Can you please rebase?

**Comment by ngoyal16** [comment]
> @ngoyal16 we [merged a fix](https://github.com/aws/aws-node-termination-handler/commit/e3258a8f60f5f58e6d1ddbbcb0a654d83842ecb4) for the failing tests, thanks for your patience. Can you please rebase?

@AustinSiu done

**Review by AustinSiu — APPROVED** [review]

**Pull request merged**